src/web/pages/permissions/__tests__/PermissionDialog.test.tsx

```diff
@@ -0,0 +1,666 @@
+/* SPDX-FileCopyrightText: 2025 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import {describe, test, expect, testing, beforeEach} from '@gsa/testing';
+import {
+  changeInputValue,
+  getSelectItemElementsForSelect,
+  screen,
+  within,
+  fireEvent,
+  rendererWith,
+} from 'web/testing';
+import Capabilities from 'gmp/capabilities/capabilities';
+import Group from 'gmp/models/group';
+import Permission from 'gmp/models/permission';
+import Role from 'gmp/models/role';
+import User from 'gmp/models/user';
+import PermissionDialog from 'web/pages/permissions/PermissionDialog';
+
+let onSave;
+let onClose;
+
+describe('PermissionDialog tests', () => {
+  beforeEach(() => {
+    onSave = testing.fn();
+    onClose = testing.fn();
+  });
+  test('should render without issues and close', () => {
+    const gmp = {};
+
+    const {render} = rendererWith({
+      gmp,
+      capabilities: true,
+      store: true,
+      router: true,
+    });
+
+    render(<PermissionDialog onClose={onClose} onSave={onSave} />);
+
+    screen.getByText('New Permission');
+
+    const close = screen.getDialogCloseButton();
+    fireEvent.click(close);
+
+    expect(onClose).toHaveBeenCalled();
+  });
+
+  test('should render with title and save', () => {
+    const gmp = {};
+
+    const {render} = rendererWith({
+      gmp,
+      capabilities: true,
+      store: true,
+      router: true,
+    });
+
+    render(
+      <PermissionDialog
+        title="Custom Permission Title"
+        onClose={onClose}
+        onSave={onSave}
+      />,
+    );
+
+    screen.getByText('Custom Permission Title');
+
+    const save = screen.getDialogSaveButton();
+    fireEvent.click(save);
+
+    expect(onSave).toHaveBeenCalledWith({
+      comment: '',
+      groupId: undefined,
+      id: undefined,
+      name: 'Super',
+      permission: undefined,
+      resourceId: '',
+      resourceName: undefined,
+      resourceType: undefined,
+      roleId: undefined,
+      subjectType: undefined,
+      title: 'Custom Permission Title',
+      userId: undefined,
+    });
+  });
+
+  test('should allow to change name and comment', async () => {
+    const gmp = {};
+
+    const testCapabilities = new Capabilities([
+      'everything',
```

Comment:
I am not sure about the intention but using everything here makes all the stuff below unnecessary.